src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -95,18 +95,18 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
             _editorFiles = new EditorFiles(this);
 
             _disposableBag
-                .Add(ProjectFiles)
-                .Add(() => Analyzer?.Dispose())
-                .Add(() => _shutdownCts.Cancel())
-                .Add(AnalysisQueue)
                 .Add(() => {
                     foreach (var ext in _extensions.Values) {
                         (ext as IDisposable)?.Dispose();
                     }
                     foreach (var ext in _oldExtensions.Values) {
                         (ext as IDisposable)?.Dispose();
                     }
-                });
+                })
+                .Add(ProjectFiles)
```

Comment:
What does this reordering affects?

Reply:
Mostly correct order - send cancellation, clear queue, then dispose analyzer, then the rest.